### ISSUE TITLE
Model works in 0.4.1 but un-simulateable in 2016-03-31

### ISSUE THREAD

**Issue by nickerso** [opencor/opencor#957]
The attached model (`tmp.cellml`) loads and is simulate-able in the 0.4.1 version of OpenCOR (OS X), but in the latest snapshot (2016-03-31) it fails with the error message: 

**/Users/dnic019/shared-folders/projects/data-2-cellml/tmp.cellml**
    **Runtime:** invalid.
    **Error:** the model must have at least one ODE or DAE.

I have created this model to try and isolate a problem which has been coming up a lot with this snapshot where models that used to simulate fine are no longer running at all. Another example, is having a complex model with lots of imports which gives the same error - but when exporting to CellML 1.0 and loading that into OpenCOR it works fine.

[tmp.cellml.zip](https://github.com/opencor/opencor/files/251520/tmp.cellml.zip)

**Comment by nickerso** [comment]
These snapshots also give the same error:

* 2016-02-01
* 2015-11-25
* 2015-11-15

These ones work:

* 2015-07-15
* 2015-09-07
* 2015-09-27

So it seems to be something in the change from 2015-09-27 to 2015-11-15 that broke it?

**Comment by agarny** [comment]
Hmm, I can indeed confirm the issue and it does indeed look like things got broken between 2015-09-27 and 2015-11-15. Ok, sorry for the problem. I am going to look into it.

**Comment by agarny** [comment]
Ok, I have just had a look at this and I have found the cause of the 'problem'. It's indirectly related to what has been done for issue #568.

Basically, to use a model in the ```Single Cell``` view, we need to get a runtime and, in the case of a CellML 1.1, we (now) build a mapping between the variables of all local components and their source (see [here](https://github.com/opencor/opencor/blob/44fdeaf1c2dc91fa059e5c75e2a6a253d31cacf3/src/plugins/support/CellMLSupport/src/cellmlfileruntime.cpp#L774)).

From there, we make sure that all the computation targets have a source that is in the main CellML file (see [here](https://github.com/opencor/opencor/blob/44fdeaf1c2dc91fa059e5c75e2a6a253d31cacf3/src/plugins/support/CellMLSupport/src/cellmlfileruntime.cpp#L808)) and it is where things 'break' for your sample model.

Indeed, for your sample model, we get the following mapping:

**Component:** data

Source variable | Variable
--- | ---
_data_ | _data_
```null``` | _time_
_dummy_ | _dummy_

So, as you can tell there is no source variable for _time_, which means that _time_ gets skipped and, as a consequence, no variable of integration is tracked (see [here](https://github.com/opencor/opencor/blob/44fdeaf1c2dc91fa059e5c75e2a6a253d31cacf3/src/plugins/support/CellMLSupport/src/cellmlfileruntime.cpp#L885)). The ultimate consequence is that the runtime is considered invalid since no variable of integration could be found, hence the 'problem' you are experiencing.

Now, looking at your sample model, it's pretty obvious why that is the case since _time_ is declared with a public interface of ```in```, i.e.

```
var time: ms {pub: in};
```

If you remove that interface declaration, everything is 'fine' (even though a variable of integration should have an initial value (if my recollection of the CellML specifications is right)).

In conclusion, I get the feeling that there is no problem. In fact, and if anything, previous versions of OpenCOR (through the use of the CellML API) are too lenient and allow something that shouldn't be allowed. Agreed? If so, feel free to close this issue. If not, then I only ask to be convinced... :)

**Comment by nickerso** [comment]
Thanks @agarny, this makes sense to me and I'm slightly embarrassed to miss such an error in the model. I agree that this is a case of the earlier versions of OpenCOR being too lenient rather than an actual error in the recent snapshots.

I'll open a new issue if the similar problem I'm seeing with more complex models is not similarly resolved as this simple test model.
